syntax = "proto3";

// Uplink messages

// robot position
message PosReport {
    float pos_x = 1;
    float pos_y = 2;
    float pos_theta = 3;
}

// robot speed, taken from motor encoders or codinf wheels
message SpeedReport {
    float vx = 1;
    float vy = 2;
    float vtheta = 3;
}

message MotorSpeedReport {
    float v1 = 1;
    float v2 = 2;
    float v3 = 3;
}

message MotorCommandReport {
    float cmd1 = 1;
    float cmd2 = 2;
    float cmd3 = 3;
}

message BatteryReport {
    float voltage = 1;
}

message UpMessage {
    oneof inner{
        PosReport pos_report = 1;
        SpeedReport speed_report = 2;
        BatteryReport battery_report = 3;
        MotorSpeedReport motor_speed_report = 4;
        MotorCommandReport motor_cmd_report = 5;
    }
}




